tox.ini

```diff
@@ -1,21 +1,23 @@
 [tox]
-envlist = {py27,py33,py34,py35,py36,pypy}-{base,optional}
+envlist = {py27,py33,py34,py35,py36,pypy}-{base,six19,optional}
 
 [testenv]
 deps =
-  flake8<3.0
-  pytest
-  pytest-expect>=1.1,<2.0
-  mock
-  base: six
-  base: webencodings
   optional: -r{toxinidir}/requirements-optional.txt
+  -r{toxinidir}/requirements-test.txt
   doc: Sphinx
 
+passenv =
+  PYTEST_COMMAND
+  COVERAGE_RUN_OPTIONS
 commands =
-  {envbindir}/py.test {posargs}
+  six19: pip install six==1.9
+  {env:PYTEST_COMMAND:{envbindir}/py.test} {posargs}
   flake8 {toxinidir}
```

Comment:
In other projects I work on, I run flake8 as a separate environment because there isn't much reason to run it multiple times. Having said that, it seems like it only takes a few seconds on html5lib-python, so seems like it'd not a big deal here.

Reply:
flake8 could be moved out to a separate factor easily, and even a different testenv.

It is important to run flake8 on each version of python, as pyflakes is version dependent.

The benefit of keeping it in here is that devs dont need to remember to do it, and they typically only test on one environment.  For CI, the running time is inconsequential compared to the rest of the very voluminous tests ;-)

Reply:
+1 to everything you say here. Don't change what you have.